llvm/lib/Target/SPIRV/SPIRVEmitNonSemanticDI.cpp

```diff
@@ -1650,26 +1584,12 @@ void SPIRVEmitNonSemanticDI::emitDebugTypePtrToMember(
       bool MemberTypeIsFwd = false;
       Register MemberTypeReg =
           findBaseTypeRegisterRecursive(BaseTy, Ctx, MemberTypeIsFwd);
-      if (!MemberTypeReg.isValid()) {
-        llvm::errs()
-            << "Warning: Could not find Member Type for PtrToMember.\n";
-        MemberTypeReg = EmitDIInstruction(
-            SPIRV::NonSemanticExtInst::DebugInfoNone, {}, Ctx, false);
-        MemberTypeIsFwd = false;
-      }
       if (MemberTypeIsFwd)
         OpHasForwardRef = true;
       const DIType *ClassTy = PtrToMemberType->getClassType();
       bool ParentTypeIsFwd = false;
       Register ParentReg =
           findBaseTypeRegisterRecursive(ClassTy, Ctx, ParentTypeIsFwd);
-      if (!ParentReg.isValid()) {
```

Comment:
Does it mean, that we keep translating the debug instruction, even if `!ParentReg.isValid()`? If backend would emit nothing in this case, then removal of these lines are fine. Otherwise, **DebugTypePtrToMember** must not be emitted in the first place. Same applicable for every instruction, that might have not-valid scope/parent operand.